app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -0,0 +1,280 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert' show jsonEncode;
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:graphql/client.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+
+import 'check_for_waiting_pull_requests_queries.dart';
+
+@immutable
+class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
+  const CheckForWaitingPullRequests(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting LoggingProvider loggingProvider,
+  })  : loggingProvider = loggingProvider ?? Providers.serviceScopeLogger,
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final LoggingProvider loggingProvider;
+
+  @override
+  Future<Body> get() async {
+    final Logging log = loggingProvider();
+    final GraphQLClient client = await config.createGitHubGraphQLClient();
+
+    await _checkPRs('flutter', 'flutter', log, client);
+    await _checkPRs('flutter', 'engine', log, client);
```

Comment:
There's no need to wait and these two functions can run concurrently.

Reply:
We should wait to avoid returning before the functions finish - we also have a lint on for this repo to avoid unawaited functions. The queries we're using don't take very long though.